doc/cephfs/client-auth.rst

```diff
@@ -153,3 +148,93 @@ The optional ``{network/prefix}`` is a standard network name and
 prefix length in CIDR notation (e.g., ``10.3.0.0/16``).  If present,
 the use of this capability is restricted to clients connecting from
 this network.
+
+File system Information Restriction
+===================================
+
+If desired, the monitor cluster can present a limited view of the file systems
+available. In this case, the monitor cluster will only inform clients about
+file systems specified by the administrator. Other file systems will not be
+reported and commands affecting them will fail as if the file systems does
+not exists.
+
+Consider following example. The Ceph cluster has 2 FSs::
+    $ ceph fs ls
+    name: cephfs, metadata pool: cephfs_metadata, data pools: [cephfs_data ]
+    name: cephfs2, metadata pool: cephfs2_metadata, data pools: [cephfs2_data ]
+
+But we authorize client ``someuser`` for only one FS::
+    $ ceph auth add client.someuser mon 'allow rw fsname=cephfs' mds 'allow rw' osd 'allow rw'
+    exported keyring for client.someuser
+    $ cat ceph.client.someuser.keyring
+    [client.someuser]
+            key = AQA5lgRfC2WvARAA+Wr7mWPwv7eXkvFByGFYJA==
+            caps mds = "allow rw"
+            caps mon = "allow rw fsname=cephfs"
+            caps osd = "allow rw"
+
+And the client can only see the FS that it has authorization for::
+    $ ceph fs ls -n client.someuser -k ceph.client.someuser.keyring
+    name: cephfs, metadata pool: cephfs_metadata, data pools: [cephfs_data ]
+
+Standby MDS daemons will always be displayed. Note that the information about
+restricted MDS daemons and file systems may become available by other means,
+such as ``ceph health detail``.
+
+MDS communication restriction
+=============================
+
+By default, user applications may communicate with any MDS, whether or not
+they are allowed to modify data on an associated file system (see
+``Path restriction`` above). Client's communication can be restricted to MDS
+daemons associated with particular file system(s) by adding MDS caps for that
+partiuclar file system::
```

Comment:
This section somewhat overlaps with the discussion on `fs authorize` at the beginning of this doc. I think you can merge them.

Reply:
Except for `fs authorize` both look pretty unrelated to each other.

Reply:
Modified `fs authorize`'s code to assign MON caps too, the doc now never uses `auth add` now.